### PR TITLE
 Upgrade to Mahout 0.9 #52

### PR DESCRIPTION
**Pull request by lewismc**
Hi @jnioche, this PR 
 * makes an attempt to upgrade to Mahout 0.9 which also means that no Hadoop Upgrade is required right now. Behemoth can exist on Hadoop 1.2.1 as it currently does. This requires updates to LuceneTokenizerMapper.java due to the change in the TokenStream API in Lucene as well as addition of the minimal document frequency score to the  HighDFWordsPruner.pruneVectors API.
 * removes unnecessary <version> elements from each of the child pom's, they already inherit this from parent pom.xml
 * generally cleans up some unused imports

This is not thoroughly tested and I am skeptical about the below proposed change

https://github.com/DigitalPebble/behemoth/compare/master...lewismc:BEHEMOTH-52?expand=1#diff-d524e30136046dca4f2964333463676aL324

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -8,7 +8,6 @@
 	</parent>
 
 	<artifactId>behemoth-core</artifactId>
-	<version>1.2-SNAPSHOT</version>
```
Comment (jnioche) [inline core/pom.xml]: This is unrelated to the Mahout upgrade and should be done separately. 
I'd rather keep the versions decoupled even if they are the same, a component could evolve independently of the core or other modules

File: mahout/src/main/java/com/digitalpebble/behemoth/mahout/SparseVectorsFromBehemoth.java [hunk 2]
```diff
@@ -39,14 +39,13 @@
...
 import org.apache.mahout.common.commandline.DefaultOptionCreator;
 import org.apache.mahout.math.hadoop.stats.BasicStats;
-import org.apache.mahout.vectorizer.DefaultAnalyzer;
 import org.apache.mahout.vectorizer.DictionaryVectorizer;
 import org.apache.mahout.vectorizer.DocumentProcessor;
 import org.apache.mahout.vectorizer.HighDFWordsPruner;
 import org.apache.mahout.vectorizer.collocations.llr.LLRReducer;
 import org.apache.mahout.vectorizer.common.PartialVectorMerger;
 import org.apache.mahout.vectorizer.tfidf.TFIDFConverter;
-import org.jfree.util.Log;
```
Comment (jnioche) [inline mahout/src/main/java/com/digitalpebble/behemoth/mahout/SparseVectorsFromBehemoth.java]: no idea where that one came from but maybe remove the commented line altogether, no point in leaving it there

**Comment by jnioche** [comment]
> removes unnecessary elements from each of the child pom's, they already inherit this from parent pom.xml

see comments

>This is not thoroughly tested and I am skeptical about the below proposed change

you mean `Class<? extends Analyzer> analyzerClass = Analyzer.class;`?

See [SparseVectorsFromSequenceFiles](https://github.com/apache/mahout/blob/mahout-0.10.x/mr/src/main/java/org/apache/mahout/vectorizer/SparseVectorsFromSequenceFiles.java) which our class should mimic as much as possible. 

BTW I see that they have release Mahout 0.10; have you tried using that version as a dependency?

**Comment by lewismc** [comment]
Hi Julien,
All of your comments are addressed bar the upgrade to Mahout 0.10.
If you want me to try that as well that I can. I'll begin work on it on the same branch and if it works out i will send another PR Julien.
Thanks

**Comment by jnioche** [comment]
thanks @lewismc! 
Mahout 0.10 \: entirely up to you. If it is straightforward then maybe worth doing.
